resources/aws/securitygroup.go

```diff
@@ -34,30 +50,51 @@ func (s SecurityGroup) findExisting() (*ec2.SecurityGroup, error) {
 				},
 			},
 		},
-	})
-	if err != nil {
-		return nil, microerror.MaskAny(err)
 	}
 
-	if len(securityGroups.SecurityGroups) < 1 {
-		return nil, microerror.MaskAny(securityGroupFindError)
+	for _, filter := range filters {
+		securityGroups, err := s.Clients.EC2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+			Filters: filter,
+		})
+		if err != nil {
+			return nil, microerror.MaskAny(err)
+		}
+
+		if len(securityGroups.SecurityGroups) < 1 {
+			continue
+		}
+
+		return securityGroups.SecurityGroups[0], nil
```

Comment:
I would like to be defensive and throw an error in case AWS does something wrong. Or it might be that we assume things that are not right or change in the future.